remix/app/routes/newsletters.tsx

```diff
@@ -0,0 +1,61 @@
+import type { User } from "@prisma/client";
+import type { LoaderArgs } from "@remix-run/node";
+import {
+  Form,
+  isRouteErrorResponse,
+  useLoaderData,
+  useRouteError,
+} from "@remix-run/react";
+import { SocialsProvider } from "remix-auth-socials";
+import { authenticator } from "~/services/auth.server";
+import type { UserProfile } from "~/types";
+import { db } from "~/utils/db.server";
+
+export const loader = async ({ request }: LoaderArgs) => {
+  const userRequest = (await authenticator.isAuthenticated(request)) as any;
+  const userProfile: UserProfile = userRequest?._json;
+  if (!userProfile) {
+    throw new Response("Forbidden", { status: 403 });
+  }
+  let user: User | null = null;
+
+  try {
+    user = await db.user.findUnique({
+      where: { email: userProfile.email },
+    });
+  } catch (err) {
+    console.log(err);
+  }
+  return { user };
+};
+
+export default function Index() {
+  const { user } = useLoaderData();
+  return (
+    <div>
+      <h1>Newsletter Page</h1>
+      <p>{user?.name}</p>
+    </div>
+  );
+}
+
+export function ErrorBoundary() {
```

Comment:
This error boundary should be define at the layout level to ensure this code is not repeated. See https://dev.to/pckilgore/remix-sharing-your-index-layout-with-sub-routes-2856

Reply:
Thank you for suggestion and the link, but it seem like don't work anymore with Remix route v2.

To avoid the repeat error boundary I have move the it to `root.tsx`.

For the next implementation I will follow the [Remix route v2](https://remix.run/docs/en/main/file-conventions/route-files-v2#nested-layouts-without-nested-urls)